python/lsst/daf/butler/core/storedFileInfo.py

```diff
@@ -252,3 +252,6 @@ def update(self, **kwargs: Any) -> StoredFileInfo:
         if kwargs:
             raise ValueError(f"Unexpected keyword arguments for update: {', '.join(kwargs)}")
         return type(self)(**new_args)
+
+    def __reduce__(self) -> str | tuple[Any, ...]:
+        return (self.from_record, (self.to_record(),))
```

Comment:
We probably should have a small round trip pickle test.

Reply:
I added tests for both pickling of StoredFileInfo and DatastoreRecordData.subset().